packaging/technical-addon/Makefile

```diff
@@ -31,89 +40,130 @@ MODINPUT_CONFIG_GENERATOR := $(TOOLS_DIR)/modinput_config_generator
 ta-build-tools:
 	@echo "Building tools..."
 	@mkdir -p $(TOOLS_DIR)
+	go build -o $(MODINPUT_CONFIG_GENERATOR) ./cmd/modinput_config_generator
 
 .PHONY: test-ta-build-tools
 test-ta-build-tools: ta-build-tools
-	cd $(SOURCE_DIR) && go build -o $(TOOLS_DIR)/modinput_config_generator ./cmd/modinput_config_generator
-	$(MODINPUT_CONFIG_GENERATOR) -source-dir=$(SOURCE_DIR)/cmd/modinput_config_generator/internal/testdata -schema-name=Sample_Addon --build-dir=$(BUILD_DIR) || exit 1; \
+	$(MODINPUT_CONFIG_GENERATOR) -source-dir=$(ADDONS_SOURCE_DIR)/cmd/modinput_config_generator/internal/testdata -schema-name=Sample_Addon --build-dir=$(BUILD_DIR) || exit 1; \
 	mkdir -p $(BUILD_DIR)/Sample_Addon/$(PLATFORM)_$(SPLUNK_ARCH)/bin; \
-	cd $(SOURCE_DIR) && go build $(GO_BUILD_FLAGS) -o $(BUILD_DIR)/Sample_Addon/$(PLATFORM)_$(SPLUNK_ARCH)/bin/Sample_Addon ./cmd/modinput_config_generator/internal/testdata/pkg/sample_addon/runner || exit 1; \
+	go build $(GO_BUILD_FLAGS) -o $(BUILD_DIR)/Sample_Addon/$(PLATFORM)_$(SPLUNK_ARCH)/bin/Sample_Addon $(ADDONS_SOURCE_DIR)/cmd/modinput_config_generator/internal/testdata/pkg/sample_addon/runner || exit 1; \
 	echo "built addon, testing..."
-	cd $(SOURCE_DIR) && go test -v ./...
+	go test -v ./...
+
+.PHONY: gen-modinput-config
+gen-modinput-config:
+	@echo "Generating runner configs..."
+	@for schema in $(MODULAR_INPUT_SCHEMAS); do \
+		schema_lower=$$(echo $$schema | tr '[:upper:]' '[:lower:]'); \
+		echo "Generating config and TA scaffold for $$schema..."; \
+		mkdir -p $(BUILD_DIR)/$$schema/; \
+		$(MODINPUT_CONFIG_GENERATOR) -source-dir=$(ADDONS_SOURCE_DIR) -schema-name=$$schema --build-dir=$(BUILD_DIR) || exit 1; \
+		ls -lAh $(BUILD_DIR); \
+	done
+
+# Build all runner binaries.. This may require some more build deps
+build-ta-runners:
+	@echo "Building runner binaries..."
+	@for schema in $(MODULAR_INPUT_SCHEMAS); do \
+		echo "Building $$schema..."; \
+		schema_lower=$$(echo $$schema | tr '[:upper:]' '[:lower:]'); \
+		for platform in $(TA_PLATFORMS); do \
+		  	if [ "$(SPLUNK_ARCH)" = "x86_64" ]; then \
+		  		export arch="amd64"; \
+		  	fi; \
+			echo "Building $${platform}_$$arch for $$schema..."; \
+			mkdir -p $(BUILD_DIR)/$$schema/$${platform}_$(SPLUNK_ARCH)/bin; \
+			GOOS=$$platform GOARCH=$$arch go build $(GO_BUILD_FLAGS) -o $(BUILD_DIR)/$$schema/$${platform}_$(SPLUNK_ARCH)/bin/$$schema $(ADDONS_SOURCE_DIR)/pkg/$$schema_lower/runner || exit 1; \
+		done \
+	done
+
+.PHONY: generate-technical-addon-linux-autoinstrumentation
+generate-technical-addon-linux-autoinstrumentation:
+	cd $(AUTOINSTRUMENTATION_DIR) && $(MAKE) dist
+	mkdir -p $(BUILD_DIR)
+	PLATFORM="all" \
```

Comment:
I would have assumed the `PLATFORM` here would be set to `linux` or something along those lines since this target is for `linux-`. Why is it set to `all`?